src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_index_pattern/components/header/header.tsx

```diff
@@ -103,14 +103,14 @@ export const Header: React.FC<HeaderProps> = ({
       <EuiFlexGroup>
         <EuiFlexItem>
           <EuiForm isInvalid={isInputInvalid}>
-            {dataSourceEnabled
+            {dataSourceEnabled && dataSourceRef?.title
```

Comment:
why can't we only check if the `dataSourceRef ` is defined, instead of using title. Is there a case that dataSouceRef  comes without title?

Reply:
Not today, but not sure in future, that's why I prefer to do the safer check  -- also the UX only uses the title section and it does not make sense to show data source when it's not presented.